app/hooks/use-create-ledger-tx.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import StacksApp, { LedgerError, ResponseSign } from '@zondax/ledger-blockstack';
+
+import { RootState } from '@store/index';
+import { STX_DERIVATION_PATH } from '@constants/index';
+import { ContractCallOptions, makeUnsignedContractCall } from '@stacks/transactions';
+import { selectPublicKey } from '@store/keys';
+import { selectCoreNodeInfo, selectPoxInfo } from '@store/stacking';
+
+interface UseCreateLedgerTxArgs {
+  txOptions: ContractCallOptions;
+  stacksApp: StacksApp;
+}
+
+export function useCreateLedgerTx() {
```

Comment:
TODO would be to have this support STX transfers too? Not blocking for this PR, just in general.

Reply:
Agree, noted